File: 1514-MinimumValueToGetPositiveStepByStepSum/1514-MinimumValueToGetPositiveStepByStepSum.py
# Last updated: 10/8/2025, 9:49:03 PM
class Solution:
    def minStartValue(self, nums: List[int]) -> int:
        
        running_sum = 0
        min_sum = 0
        for num in nums:
            running_sum += num
            min_sum = min(min_sum, running_sum)
        return 1 - min_sum
        

File: 1514-MinimumValueToGetPositiveStepByStepSum/1514-MinimumValueToGetPositiveStepByStepSum.java
// Last updated: 10/8/2025, 9:49:06 PM
class Solution {
    public int minStartValue(int[] nums) {
        int startValue=1;
        int total=0;
        
        while(true){
            total=startValue;
            boolean isValid=true;
            for(int i=0;i<nums.length;i++){
                total+=nums[i];
                if(total<1){
                    isValid=false;
                    break;
                }
            }
            if(isValid){
                return startValue;
            }
            else{
                startValue=startValue+1;
            }
        }
    }
}